### ISSUE TITLE
Wind spectrum - macro atom models

### ISSUE THREAD

**Issue by kslong** [sirocco-rt/sirocco#739]
We would like to be able to compare the spectrum of the wind in macro-atom models to those in simple atom models.  This issue is about how we go about making this comparison.

The spectrum of the wind in a macro atom model is the sum of all the photons  produced by macro-atom interactions as photons are transiting the wind.  This spectrum has to be created on the fly as these interactions take place.  (This takes place around line 1250 in resonate.c).   

The other spectra are accumulated at the end of ionization or spectrum cycles by iterating all of the photon bundles in spectra.c  Currently, in spectra.c, we do accrue spectra for wind photons in the simple_atom case, but the weights we store there are the observed weights, but not the initial weights, so this would not provide a direct comparison.  This is useful since it does show how much of the wind is self-absorbed, but it does not allow for a direct comparison.

There are several options:

- We could add the photons "created" in the macro atoms case to the "CREATED" photon column of the output files. These do have all of the original weights at least. This does not isolate the wind photons.
- We could add some additional columns to all of the spectra (one that consists of the created values and one that consists of the extracted values. The main downside of this is that it adds columns to our spectral tables.  (It's not obvious in this case how to compare "observed wind photons", because we treat different processes as scattering processes, but this could be worked out).

@Higginbottom   do you or anyone have thoughts on this.

Aside:  There is a separate small issue to be dealt with at the same time as we deal with this. Right now when we are in the reflect photons mode, we record no photons as hitting the disk.  This is the same kind of problem as for wind photons in the macro atom mode, in the sense the the spectrum needs to be accumulated on the fly.

**Comment by kslong** [comment]
The need to address this is really associated with #659  (the problem we colloquially refer to as Mandy's problem), or more generically for understanding differences between macro-atom and simple atom models.

**Comment by kslong** [comment]
In looking at this problem, I have noticed that we accumulate spectra both logarithmic and linear which are labeld f_wind and lf_wind in addition to the normal versions.  

In python.h it says

````
 double f_wind[NWAVE];         /* The spectrum of photons created in the wind or scattered in the wind. Created for 
                                   reflection studies but possibly useful for other reasons as well. */
  double lf_wind[NWAVE];        /* The logarithmic version of this */
````

As far as I can tell these were not printed out.  It's possible these were created by me in an abortive attempt  to isolate a reflection spectrum but I am tempted to delete them for now.  Do others agree.

--- 

this last statement was incorrect, the wind photons are in some sense spectra created from wind photons, and they are printed out.   I suspect but do not know that @Higginbottom added these at some point.  I am agnostic on whether they should remain, but I think we need a discussion on what their utility is.

**Comment by kslong** [comment]
It was not difficult to add photons created by macro-atom interactions to the "Created" spectrum.  This does not involve changing any of the output spectral files.  However, I'm seriously considering splitting the Created spectrum into the spectrum created by external sources and that created in the wind.  For the macro-atom case this would the photons created by macro-atom interactions; for the simple case it would simple be the wind photons created at the beginning of each cycle.  Comments welcome, but I am moving on this now, so earlier is better than later.

**Comment by kslong** [comment]
OK, I have largely completed work on this problem.  I have added a column to the spectrum files called WCreated, which contains the spectra of photons which were created in the wind.  The Created column now only contains those photons that were created from external sources.  (All of the spectral files, including the ones specific to the wind have this problem).  

As part of doing this, we now capture the photons that hit a surface (even if they are reflected).  

I have also modified the routines used to plot results for the regression tests, and for the run_checks.  The regression test outputs only change in minor ways, but the run_check plots look different now.
